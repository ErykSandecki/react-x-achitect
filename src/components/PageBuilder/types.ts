// types
import { TRectCoordinates } from 'types';

export type TRectArea = TRectCoordinates | null;
