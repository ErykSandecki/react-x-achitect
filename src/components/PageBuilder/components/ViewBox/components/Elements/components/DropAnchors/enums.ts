export enum DropAnchorsPosition {
  top = 'top',
  bottom = 'bottom',
  left = 'left',
  right = 'right',
}
