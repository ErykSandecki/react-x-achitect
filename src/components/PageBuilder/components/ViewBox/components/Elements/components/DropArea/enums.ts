export enum DropAreaPosition {
  top = 'top',
  bottom = 'bottom',
  left = 'left',
  right = 'right',
}
