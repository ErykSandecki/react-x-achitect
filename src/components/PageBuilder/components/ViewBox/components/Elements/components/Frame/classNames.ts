export const className = 'Frame';

export const classNames = {
  [className]: className,
  label: {
    name: `${className}__label`,
    modificators: {
      selected: `${className}__label--selected`,
    },
  },
};
