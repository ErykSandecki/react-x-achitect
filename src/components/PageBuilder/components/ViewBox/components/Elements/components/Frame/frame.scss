@import '../../../../../../styles/z-indexes.scss';

.Frame {
  @include theme-selector('blue-1', color, '&:hover .Frame__label');

  background-color: white;

  &:hover {
    z-index: $--rx-zindex-element-active;
  }

  &__label {
    @include theme('neutral-2', color);

    font-size: 11px;
    position: absolute;
    transform: translate(1.5px, -20px);

    &--selected {
      @include theme('blue-1', color);
    }
  }
}
