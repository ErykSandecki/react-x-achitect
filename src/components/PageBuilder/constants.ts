export const translationNameSpace = 'pageBuilder';

export const TOOLBAR_HEIGHT = 56;
