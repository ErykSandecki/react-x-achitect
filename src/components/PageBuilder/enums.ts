export enum Constrain {
  bottom = 'bottom',
  centerHorizontal = 'centerHorizontal',
  centerVertical = 'centerVertical',
  left = 'left',
  right = 'right',
  top = 'top',
}
