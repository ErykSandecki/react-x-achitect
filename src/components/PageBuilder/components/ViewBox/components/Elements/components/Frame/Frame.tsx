import { FC, memo } from 'react';
import { useTranslation } from 'react-i18next';

// components
import MoveableElement from '../MoveableElement/MoveableElement';
import { Small } from 'shared';

// hooks
import { useTheme } from 'hooks';

// others
import { className as classNameFrame, classNames } from './classNames';
import { translationNameSpace } from './contants';

// styles
import styles from './frame.scss';

// types
import { TElementProps } from '../../types';

export type TFrameProps = TElementProps & {
  id: string;
};

const Frame: FC<TFrameProps> = ({ className, id, mouseMode }) => {
  const { classNamesWithTheme, cx } = useTheme(classNames, styles);
  const { t } = useTranslation();

  return (
    <MoveableElement
      classes={{
        className: cx(className, classNamesWithTheme[classNameFrame]),
      }}
      id={id}
      mouseMode={mouseMode}
    >
      <Small classes={{ className: cx(classNamesWithTheme.label) }}>
        {t(`${translationNameSpace}.label.createFrame`)}
      </Small>
    </MoveableElement>
  );
};

export default memo(Frame);
