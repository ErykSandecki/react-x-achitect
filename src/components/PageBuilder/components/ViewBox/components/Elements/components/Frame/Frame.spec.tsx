import { Provider } from 'react-redux';
import { render } from '@testing-library/react';

// components
import Frame from './Frame';

// mocks
import {
  elementDynamicDataMock,
  elementStaticDataMock,
  pageBuilderStateMock,
} from 'test/mocks/reducer/pageBuilderMock';
import { wholeStateMock } from 'test/mocks/reducer/wholeStateMock';
import 'test/mocks/sagas/allSagas';

// others
import { REDUCER_KEY as PAGE_BUILDER } from 'store/pageBuilder/actionsType';

// store
import { configureStore } from 'store/store';

// types
import { MouseMode } from 'components/PageBuilder/enums';

const stateMock = {
  ...wholeStateMock,
  [PAGE_BUILDER]: {
    ...pageBuilderStateMock[PAGE_BUILDER],
    pages: {
      ...pageBuilderStateMock[PAGE_BUILDER].pages,
      ['0']: {
        ...pageBuilderStateMock[PAGE_BUILDER].pages['0'],
        elements: {
          dynamicData: { [elementDynamicDataMock.id]: elementDynamicDataMock },
          staticData: { [elementStaticDataMock.id]: elementStaticDataMock },
        },
      },
    },
  },
};

describe('Frame snapshots', () => {
  it('should render Frame', () => {
    // mock
    const store = configureStore(stateMock);

    // before
    const { asFragment } = render(
      <Provider store={store}>
        <Frame
          className="className"
          id={elementDynamicDataMock.id}
          mouseMode={MouseMode.default}
          parentId={elementStaticDataMock.parentId}
          type={elementStaticDataMock.type}
        />
      </Provider>,
    );

    // result
    expect(asFragment()).toMatchSnapshot();
  });
});
