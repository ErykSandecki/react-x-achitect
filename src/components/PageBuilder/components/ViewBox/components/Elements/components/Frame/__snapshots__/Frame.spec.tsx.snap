// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Frame snapshots should render Frame 1`] = `
<DocumentFragment>
  <div
    class="css-q0izro className Frame Frame--light Element Element--light"
    data-status="false"
    data-test-box=""
    id="1"
    style="background-color: rgb(255, 255, 255); height: 100px; left: 0px; position: absolute; top: 0px; transform: rotate(0deg); width: 100px;"
  >
    <small
      class="css-q0izro Frame__label Frame__label--light Typography Typography--light Typography--neutral1 Typography--neutral1--light Typography--regular Typography--regular--light Typography--normal Typography--normal--light Typography--small Typography--small--light"
      data-test-text=""
      style="text-align: inherit;"
    >
      Frame
    </small>
  </div>
</DocumentFragment>
`;
